ethcore/src/client/client.rs

```diff
@@ -122,11 +122,13 @@ impl SleepState {
 /// Call `import_block()` to import a block asynchronously; `flush_queue()` flushes the queue.
 pub struct Client {
 	mode: Mode,
-	chain: Arc<BlockChain>,
-	tracedb: Arc<TraceDB<BlockChain>>,
+	chain: RwLock<Arc<BlockChain>>,
```

Comment:
Yeah, I understand, but the point is that even if you replace the pointer someone might be using the old one anyway (because after getting a lock you can just clone the `Arc` and release the lock).